### PR TITLE
ARXIVNG-1463: adding Submission.compilations: List[Compilation]

### PR DESCRIPTION
**Pull request by JaimieMurdock**
Some scaffolding to enable further development on submission-ui. 

Changes:
1. The addition of the `Compilation` class, which is stored on a `Submission` object via `Submission.compilations`.
2. Renaming of `SubmissionCompiled` to `SubmissionProduct` which more accurately describes what that class captures.

Discussions:
1. Is `SubmissionProduct` even necessary now?
2. Does `Compilation.status` need to have an explicit enumeration of statuses like `Submission` and `UserRequest`?

### CODE REVIEW

File: core/arxiv/submission/domain/submission.py [hunk 1]
```diff
@@ -79,6 +79,17 @@ class SubmissionCompiled:
...
 
+@dataclass
+class Compilation:
+    """Represents a submission compilation."""
+
+    task_id: str
+    source_etag: str
+    format: str
+    start_time: datetime = field(default_factory=get_tzaware_utc_now)
+    status: str
```
Comment (erickpeirson) [inline core/arxiv/submission/domain/submission.py]: Consider adding a default status? Also, what values are acceptable? (consider an Enum type)

File: core/arxiv/submission/domain/submission.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ class SubmissionContent:
 
 
 @dataclass
-class SubmissionCompiled:
+class SubmissionProduct:
```
Comment (erickpeirson) [inline core/arxiv/submission/domain/submission.py]: Indeed, I'm not sure what we need this class for, given the introduction of  ``Compilation``

**Comment by DavidLFielding** [comment]
I'm just browsing several of the open PRs. Trying to catch up on what's happening in core.

I take it we are not interested in when a compilation finishes. Will we ever care about statistics on how long compilations take?

Is format what we currently refer to as source_format in legacy system? Are expected values 'tex' or 'pdftex'.

I'm jumping in here so I'm likely missing a lit of context. If you have a status where the compilation failed are the errors stored somewhere else? Or should they be stored with the Compilation?

File: core/arxiv/submission/domain/submission.py (hunk 1 above)
Comment (JaimieMurdock) [reply core/arxiv/submission/domain/submission.py]: Made these mirror the status in compiler.domain: https://github.com/arXiv/arxiv-compiler/blob/develop/compiler/domain.py#L33

**Comment by JaimieMurdock** [comment]
I've resolved all the issues with refactoring `compiled_content` to `compilations, except the schema file `submission.json`, which I don't have the expertise to resolve - I'm not sure if just renaming really captures the changes on the API side, since `source_content` and `compiled_content` share the same `content.json` definition.

https://github.com/arXiv/arxiv-submission-core/blob/develop/metadata/schema/resources/submission.json

**Pull request merged**